packager/mpd/base/period.cc

```diff
@@ -112,12 +112,26 @@ AdaptationSet* Period::GetOrCreateAdaptationSet(
 
     for (AdaptationSet* adaptation_set : adaptation_sets) {
       if (protected_adaptation_set_map_.Switchable(*adaptation_set,
-                                                   *new_adaptation_set)) {
+                                                   *new_adaptation_set, false)) {
```

Comment:
The code here shouldn't need to be updated if you don't plan to refactor ProtectedAdaptationSetMap out.

Instead, what you need is to change line 85 to include the bool flag.